dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,226 @@
+import logging
+import json
+import requests
+
+from funcx import ContainerSpec
+from time import sleep
+from github import Github
+import base64
+
+import mdf_toolbox
+
+
+logger = logging.getLogger(__name__)
+
+
+def create_container_spec(metadata):
+    """
+    Create the container spec for the Container Service. Iterate through
+    the dependencies and add them. Also include parsl, toolbox, home_run, etc.
+
+    :param metadata:
+    :return:
+    """
+
+    # Get the list of requirements from the schema
+    dependencies = []
+    try:
+        for k, v in metadata['dlhub']['dependencies']['python'].items():
+            dependencies.append("{0}=={1}".format(k, v))
+    except:
+        # There are no python dependencies
+        pass
+
+    model_location = None
+    # If the model was uploaded using a signed URL, it will have a
+    # transfer_method of S3 in its metadata.
+    # If the model is being built from a repo, it will not have a
+    # transfer_method, but will have a 'repository'
+    if 'transfer_method' in metadata['dlhub']:
+        if 'S3' in metadata['dlhub']['transfer_method']:
+            model_location = metadata['dlhub']['transfer_method']['S3']
+    if 'repository' in metadata:
+        model_location = metadata['repository']
+    if model_location is None:
+        raise Exception("No model location exists in metadata")
+
+    cs = ContainerSpec(
+        name=metadata['dlhub']['shorthand_name'],
+        pip=dependencies,
+        python_version="3.7",
+        payload_url=model_location,
+        conda=[],
+    )
+
+    return cs
+
+
+def check_container_build_status(funcx_client, container_uuid):
+    # Set timeout at 10 minutes
+    timeout_at = 600
```

Comment:
I wonder if this should be configurable. Are we confident 10m is enough? I have definitely seen containers take toward 10 mins to build.

Reply:
Thanks Ryan! Let's reflect that -- I thought 10min was safe but I guess not, whoops!